python/taichi/lang/kernel_impl.py

```diff
@@ -203,10 +203,11 @@ def func_call_rvalue(self, key, args):
         # Skip the template args, e.g., |self|
         assert self.is_real_function
         non_template_args = []
-        for i, anno in enumerate(self.argument_annotations):
-            if not isinstance(anno, template):
-                if id(anno) in primitive_types.type_ids:
-                    non_template_args.append(ops.cast(args[i], anno))
+        for i, args in enumerate(self.arguments):
+            if not isinstance(args.annotation, template):
+                if id(args.annotation) in primitive_types.type_ids:
+                    non_template_args.append(ops.cast(args[i],
+                                                      args.annotation))
```

Comment:
You may want to use something other than `args` as the loop variable since this function (`func_call_rvalue(self, key, args)`) already has `args` as one of its argument.

See also https://github.com/taichi-dev/taichi/runs/5964686941?check_suite_focus=true#step:6:9364

Reply:
Yep, this is a silly mistake cause I didn't run tests last time, sry for that.